partitioner/parsers/open_street_map.py

```diff
@@ -28,3 +28,44 @@ def get_node_info(self, line: Text) -> Tuple[int, float, float]:
 
     def get_edge_info(self, line: Text) -> Tuple[int, int, float]:
         pass
+
+    def is_in_partition(lon, lat, partition):
+        x_min, x_max = partition[0]
+        y_min, y_max = partition[1]
+        return x_min < lon <= x_max and y_min < lat <= y_max
+
+    def parse_way(way_lines):
+        tag_regex = re.compile(r'\s*<tag k="(\w+)" v="(.+)"\s*\/>')
+        node_regex = re.compile(r'\s*<nd ref="(\d+)"\s*\/>')
+        way_begin = re.compile(r'\s*<way [^>]*>\s*')
+        way_end = re.compile(r'\s*<\/way>\s*')
+
+        assert way_begin.match(way_lines[0])
+        assert way_end.match(way_lines[-1])
+
+        tag_dict = {}
+        node_list = []
+        for line in way_lines:
+            if node_regex.match(line):
+                match = node_regex.match(line)
+                node_list.append(int(match.group(1)))
+            elif tag_regex.match(line):
+                match = tag_regex.match(line)
+                tag_dict[match.group(1)] = match.group(2)
+
+        return node_list, tag_dict
+
+    def get_partition_nodes(self, partitions: List[List[Tuple[float]]], partition_ix: int) -> Dict[int, Tuple[float, float]]:
+        partition = partitions[partition_ix]
+        x_min, x_max = partition[0]
+        y_min, y_max = partition[1]
```

Comment:
są ustawione te miny i maxy ale z tego co widzę to nie użyte w tej pętli niżej, raczej powinien być filtr is_in_partition